2022Swerve/src/main/java/frc/robot/subsystems/AutonSubsytem.java

```diff
@@ -0,0 +1,137 @@
+package frc.robot.subsystems;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.Trajectory;
+import edu.wpi.first.math.trajectory.TrajectoryUtil;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.util.datalog.DataLog;
+import edu.wpi.first.util.datalog.StringLogEntry;
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.Filesystem;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.SwerveControllerCommand;
+import edu.wpi.first.wpilibj2.command.WaitCommand;
+import frc.robot.Constants.AutonConstants;
+import frc.robot.Constants.DriveConstants;
+
+
+public class AutonSubsytem extends SubsystemBase{
+    private enum AutonModes{
+        FORWARD, // Go forward 2 meters
+        BACKWARD, // Wait 3 seconds, go backward 2 meters
+        FORWARD180; // Go forward 2 meters and rotate 180 degrees
+    }
+
+    private ShuffleboardTab configTab = Shuffleboard.getTab("Config");
+    private NetworkTableEntry delayEntry = configTab.add("Auton Delay", 0.0).getEntry();
+    private SendableChooser<AutonModes> modeChooser = new SendableChooser<>();
+
+    private DataLog datalog = DataLogManager.getLog();
+    private StringLogEntry trajectoryLog = new StringLogEntry(datalog, "/auton/trajectory"); //Logs x translation state output
```

Comment:
What else are we logging?  What else would you need if something went wrong?  If you had a method for the wait, and method for align ( Mentioned elsewhere ) , then you could possibly log those also.

The delayEntry that is used to wait at the beginning seems important to log.  If we have a weird delay, it would be good to know if ShuffleBoard messed up and we got the wrong number...and waited correctly...Or if we did not get a delay from shuffleboard and waited erroneously somehow

Reply:
I'll make a wait method, with logging